client/my-sites/invites/invite-accept-logged-out/index.jsx

```diff
@@ -56,25 +54,24 @@ class InviteAcceptLoggedOut extends React.Component {
 		this.setState( { submitting: true } );
 		debug( 'Storing invite_accepted: ' + JSON.stringify( invite ) );
 		store.set( 'invite_accepted', invite );
-		const createAccountCallback = ( error, bearerToken ) => {
-			debug( 'Create account error: ' + JSON.stringify( error ) );
-			debug( 'Create account bearerToken: ' + bearerToken );
-
-			if ( error ) {
-				store.remove( 'invite_accepted' );
-				this.setState( { submitting: false } );
-			} else if ( bearerToken ) {
-				this.setState( { bearerToken, userData } );
-			}
-		};
 
 		const enhancedUserData = { ...userData };
 
 		if ( get( invite, 'site.is_wpforteams_site', false ) ) {
 			enhancedUserData.signup_flow_name = 'p2';
 		}
 
-		this.props.createAccount( enhancedUserData, invite, createAccountCallback );
+		this.props
+			.createAccount( enhancedUserData, invite )
+			.then( ( response ) => {
+				const bearerToken = response.bearer_token;
+				debug( 'Create account bearerToken: ' + bearerToken );
+				this.setState( { bearerToken, userData } );
+			} )
+			.catch( () => {
+				store.remove( 'invite_accepted' );
```

Comment:
Uber nit: we could have preserved the `debug( 'Create account error: ' + JSON.stringify( error ) );` here. If we want to remove it, perhaps we should also remove the success debug line too?

Reply:
Added the `debug` back. Although I personally would never look at this particular message -- the REST error is clearly visible in the Network devtools tab in a much more accessible form.

Reply:
For the protocol, I never use the `debug` calls either. I'd be totally fine with removing all of them 😉  My comment was just in the interest of preserving everything that was there before, but feel free to ditch it if you like.

Reply:
I use them maybe once a year 🙂